pageserver/src/config.rs

```diff
@@ -1359,6 +1363,32 @@ trace_read_requests = {trace_read_requests}"#,
         Ok(())
     }
 
+    #[test]
+    fn parse_incorrect_tenant_config() -> anyhow::Result<()> {
+        let tempdir = tempdir()?;
+        let (workdir, pg_distrib_dir) = prepare_fs(&tempdir)?;
+
+        let config_string = format!(
+            r#"
+        {ALL_BASE_VALUES_TOML}
+        pg_distrib_dir='{}'
+        [tenant_config]
+        checkpoint_distance = -1 # supposed to be an u64
+    "#,
+            pg_distrib_dir.display()
+        );
+
+        let toml = config_string.parse()?;
+
+        let error = PageServerConf::parse_and_validate(&toml, &workdir).unwrap_err();
+
+        assert!(error
+            .to_string()
+            .contains("checkpoint_distance: invalid value: integer `-1`, expected u64"));
```

Comment:
What does the error say before making the path_to_error changes?

Could you make this a full string comparison instead?